### PR TITLE
Annotation filter doc

### PR DESCRIPTION
**Pull request by soyuka**
@meyerbaptiste where are the docs for Property and Group filters?

### CODE REVIEW

**Comment by meyerbaptiste** [comment]
Here @soyuka: https://github.com/api-platform/docs/blob/2.1/core/filters.md#serializer-filters

Why? 😃

**Comment by soyuka** [comment]
I'm stupid, mb I think I didn't based my commit on the origin/master.

**Comment by bendavies** [comment]
@soyuka think you committed from using the wrong git user

**Comment by soyuka** [comment]
I know, wasn't my computer and lazy to git config on a temporary hardware. Doesn't change anything as I'm waiting for reviews and will probably edit/rebase with my correct one.

**Comment by soyuka** [comment]
WDYT?

**Comment by soyuka** [comment]
I've updated the filters doc, may you take a look?

File: core/filters.md [hunk 1]
```diff
@@ -285,27 +284,20 @@ Syntax: `?order[property]=<asc|desc>`
...
 namespace AppBundle\Entity;
 
+use ApiPlatform\Core\Annotation\ApiFilter;
 use ApiPlatform\Core\Annotation\ApiResource;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\OrderFilter;
 
 /**
- * @ApiResource(attributes={"filters"={"offer.order_filter"}})
+ * @ApiResource
+ * @ApiFilter(OrderFilter::class, properties={"id", "name"})
```
Comment (soyuka) [inline core/filters.md]: I need to review this as this will not work yet.

**Comment by sroze** [comment]
It's conflicting now 🤷‍♂️

**Comment by soyuka** [comment]
Yes because I changed the whole docs for annotation filters :). Should I rebase?

**Comment by dunglas** [comment]
Can you rebase @soyuka? Feel free to merge it :)

**Comment by soyuka** [comment]
It's a mess to rebase oO

**Comment by soyuka** [comment]
Because it's a lot of work to rebase I'm merging this so that next filter changes occurs on the new version. Because it's master I think it's okay if not we can still revert :).  

Done:
- changed every examples to use the `ApiFilter` annotation
- added back the note on how the annotation works
- simplified Property, Group filter options enumeration

**Pull request merged**